src/Network/Transport/InMemory.hs

```diff
@@ -62,17 +67,26 @@ data LocalConnectionState
   | LocalConnectionClosed
   | LocalConnectionFailed
 
+newtype TransportInternals = TransportInternals (TVar TransportState)
```

Comment:
Needs to be exported too. Otherwise expose internals useless. Needs a docstring. To the effect of "Internal state of the transport."